autogen/agentchat/agent.py

```diff
@@ -1,51 +1,62 @@
-from typing import Dict, List, Optional, Union
+from typing import Any, Callable, Dict, List, Optional, Protocol, Union, runtime_checkable
 
 
-class Agent:
-    """(In preview) An abstract class for AI agent.
+@runtime_checkable
+class Agent(Protocol):
+    """(In preview) A protocol for Agent.
 
     An agent can communicate with other agents and perform actions.
     Different agents can differ in what actions they perform in the `receive` method.
     """
 
-    def __init__(
-        self,
-        name: str,
-    ):
-        """
-        Args:
-            name (str): name of the agent.
-        """
-        # a dictionary of conversations, default value is list
-        self._name = name
-
     @property
-    def name(self):
-        """Get the name of the agent."""
-        return self._name
+    def name(self) -> str:
+        """The name of the agent."""
+        ...
 
-    def send(self, message: Union[Dict, str], recipient: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract method) Send a message to another agent."""
+    @property
+    def description(self) -> str:
+        """The description of the agent. Used for the agent's introduction in
+        a group chat setting."""
+        ...
 
-    async def a_send(self, message: Union[Dict, str], recipient: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract async method) Send a message to another agent."""
+    def send(
+        self,
+        message: Union[Dict[str, Any], str],
+        recipient: "Agent",
+        request_reply: Optional[bool] = None,
+    ) -> None:
+        """Send a message to another agent."""
+        ...
 
-    def receive(self, message: Union[Dict, str], sender: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract method) Receive a message from another agent."""
+    async def a_send(
+        self,
+        message: Union[Dict[str, Any], str],
```

Comment:
When using Any in Dict[str, Any], it makes hard to create fungible agents.  I realize this is not introduced in this PR. 
 I am curious, what's the reason for Any?

Reply:
`Any` is used because the dictionary could contain nested dictionaries with unspecified value types. I personally would love to have more strong enforcement of types but for (1) the frequent API changes, and (2) the frequent underlying model changes, we don't want to make any promise regarding API stability right now and avoid pre-maturely make design choices. @sonichi do you want to input on this?

Reply:
Resolve this as it has been explained in walk-through